use_camera: both
arm_topic: arm_controller/command
color_info_topic_name: /realsense/color/camera_info
color_topic_name: /realsense/color/image_raw
depth_topic_name: /realsense/aligned_depth_to_color/image_raw
points_topic_name: /realsense/depth/points
kinect_color_topic_name: /kinect/color/image_rect_color
kinect_depth_topic_name: /kinect/depth_to_color/image_raw
kinect_points_topic_name: /kinect/depth/points
transform_from_frame: world
graspnet_checkpoint_path: src/ocrtoc_perception/graspnet/logs/log_rs/checkpoint.tar
both_camera_fix_pose_file: config/arm_poses_Franka_realsense.csv
realsense_camera_fix_pose_file: config/arm_poses_Franka_realsense.csv
# Select a pose estimation method [icp|superglue]
pose_method: superglue

reconstruction:
    x_min: -0.20
    y_min: -0.6
    z_min: 0.0
    # z_min: -0.05
    x_max: 0.30
    y_max: 0.6
    z_max: 0.4
    nb_neighbors: 50
    std_ratio: 2.0
    voxel_size: 0.0015
    icp_max_try: 5
    icp_max_iter: 2000
    translation_thresh: 3.95
    rotation_thresh: 0.02
    max_correspondence_distance: 0.02
    both_camera_order: [1,0,2] #[2,1,0,3,4,5,6]
    realsense_camera_order: [2,1,0,3,4,5]

6d_pose_by_geometry:
    models_package: ocrtoc_materials
    models_dir: models
    voxel_size: 0.003
    statistical_outlier:
        nb_neighbors: 10
        std_ratio: 0.5
    segment_plane:
        distance_threshold: 0.005
        ransac_n: 20
        num_iterations: 50
    radius_outlier:
        nb_points: 400
        radius: 0.05
    template:
        num_views: 50
        num_angles: 12
    cluster_dbscan:
        eps: 0.01
        min_points: 100
        print_progress: True
    icp:
        max_correspondence_distance: 0.0024
    z_min_bound: 0.01

superglue:
    models_package: ocrtoc_materials
    models_dir: models
    rendered_object_dir: src/ocrtoc_perception/pose/rendered_object_images
    img_h: 720
    img_w: 1280
    realsense_radius: 0.4
    superpoint:
        nms_radius: 4
        keypoints_threshold: 0.005
        max_keypoints: 1024
    superglue:
        weights: outdoor
        sinkhorn_iterations: 20
        match_threshold: 0.2
    compute_pose:
        resize: [640, 360]
        device: 'cuda'
        correspondence_thresh: 10
        min_correspondence_thresh: 5

graspnet:
    refine_approach_dist: 0.081

response:
    dist_thresh: 0.1
    angle_thresh: 20
    mask_thresh: 0.5

debug: False
debug_pointcloud: True

contact_graspnet: True
6dof_graspnet: False
